examples/custom-assessment/src/main.py

```diff
@@ -0,0 +1,199 @@
+"""
+Main script for assessing an agent with custom initial conditions on Azure ML.
+
+You can test this code locally by launching the script with the flag
+``--test-local``. This will run the script locally using Ray.
+
+Note that RLlib Algorithm instances are not serializable by Ray. Therefore,
+we cannot use ray.put() to pass them as arguments to remote functions.
+Instead, we will use RLlib’s checkpointing mechanism to save and load the
+Algorithm state for each worker.
+"""
+import argparse
+import json
+import os
+import sys
+from pathlib import Path
+from typing import Dict
+
+import numpy as np
+import pandas as pd
+import ray
+from ray.rllib.algorithms.algorithm import Algorithm
+from ray.tune.registry import register_env
+from ray_on_aml.core import Ray_On_AML
+from sim import (
+    SimpleAdder as SimEnv,  # IMPORTANT: Change to import your own environment
+)
+
+# Disable deduplication of Ray logs.
+os.environ["RAY_DEDUP_LOGS"] = "0"
+
+# IMPORTANT: sim_env must be equal to the name used to register the environment when
+# training the agent.
+register_env("sim_env", lambda conf: SimEnv(conf))
+
+
+@ray.remote
+class EpisodeRunner:
+    """Class to run an episode of the simulation using a trained agent's policy."""
+
+    def __init__(
+        self,
+        checkpoint_folder: str,
+    ):
+        """Initialize the episode runner by instantiating the agent.
+
+        Args:
+            checkpoint_folder (str): The name of the folder where the checkpoint is
+            located.
+        """
+        # Instantiate the trained agent from the checkpoint folder
+        checkpoint_path = Path(__file__).absolute().parent.parent / checkpoint_folder
+        self.agent = Algorithm.from_checkpoint(checkpoint_path)
+
+        # Initialize the simulation environment
+        # IMPORTANT: Change to import your own environment
+        self.sim = SimEnv(env_config={})
+
+    def episode_to_df(
+        self, states, actions, rewards, terminated, truncated, iterations
+    ) -> pd.DataFrame:
+        """Create Pandas dataframe from episode iterations.
+
+        Args:
+            states (dict): The states of the simulation at each iteration.
+            actions (dict): The actions of the agent at each iteration.
+            rewards (list): The rewards of the agent at each iteration.
+            terminated (list): Whether the episode terminated at each iteration.
+            truncated (list): Whether the episode was truncated at each iteration.
+            iterations (int): The final number of iterations in the episode.
+
+        Returns:
+            pd.DataFrame: The dataframe containing the states, actions, episode id,
+            iteration number, and initial condition.
+        """
+        df = pd.concat([pd.DataFrame(dict) for dict in [states, actions]], axis=1)
```

Comment:
Shall we prepend ``action_`` and ``state_`` to the related variables, similarly as you do with config?